08 - Task 11: Building zephyr on atsamr21_xpro not possible

Looks like the sRAM is to small.

For recreating this command is used in folder zephyrproject/zephyr/samples/net:
west build -b atsamr21_xpro ./sockets/echo_server -- -DOVERLAY_CONFIG=overlay-802154.conf

this is the output from the terminal:

`-- Configuring done
-- Generating done
-- Build files have been written to: /home/laptop/zephyrproject/zephyr/samples/net/build
-- west build: building application
[1/220] Preparing syscall dependency handling

[2/220] Generating include/generated/version.h
-- Zephyr version: 3.3.0-rc1 (/home/laptop/zephyrproject/zephyr), build: v3.3.0-rc1-32-g48a6f160f285
[210/220] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/laptop/zephyrproject/zephyr/samples/net/build/zephyr/zephyr_pre0.map 
: && ccache /home/laptop/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/laptop/zephyrproject/zephyr/samples/net/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ieee802154/libsubsys__net__l2__ieee802154.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/config/libsubsys__net__lib__config.a  zephyr/subsys/net/lib/conn_mgr/libsubsys__net__lib__conn_mgr.a  zephyr/subsys/random/libsubsys__random.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/ieee802154/libdrivers__ieee802154.a  zephyr/drivers/spi/libdrivers__spi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/home/laptop/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/thumb/v6-m/nofp"  -L/home/laptop/zephyrproject/zephyr/samples/net/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m0plus  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/laptop/zephyrproject/zephyr/samples/net/build/zephyr && /usr/bin/cmake -E echo
/home/laptop/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `noinit' will not fit in region `RAM'
/home/laptop/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: region `RAM' overflowed by 1456 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/laptop/zephyrproject/zephyr/samples/net/build
`

--- a/08-interop/README.md
+++ b/08-interop/README.md
@@ -200,14 +200,20 @@ node running Zephyr.
 
 #### Setting up the Zephyr node
 
+> [!CAUTION]
+> Zephyr appears to be too big nowadays to compile the simple echo shell
+> for the `samr21-xpro`, it can not fit into the RAM anymore.
+> Use another board, such as the `nrf52840dk/nrf52840`
+
 1. Follow the
-['Getting Started'](https://docs.zephyrproject.org/latest/getting_started/index.html)
+['Getting Started'](https://docs.zephyrproject.org/latest/develop/getting_started/index.html)
 guide on Zephyr's documentation.
 2. You will be using the `echo_server` example application. To compile and
    flash, connect the samr21-xpro board and in zephyr's root directory run:
 
    ```sh
-   west build -p auto -b atsamr21_xpro samples/net/sockets/echo_server -- -DOVERLAY_CONFIG=overlay-802154.conf
+   west update
+   west build -p auto -b nrf52840dk/nrf52840 samples/net/sockets/echo_server -- -DOVERLAY_CONFIG=overlay-802154.conf
    west flash
    ```
 3. If everything is OK you should be able to connect to the serial output use
@@ -219,7 +225,7 @@ guide on Zephyr's documentation.
    node's IPv6 by running the following on its shell:
 
    ```sh
-   # On samr21-xpro:zephyr node
+   # On nrf52840dk/nrf52840:zephyr node
    uart:~$ net ipv6
 
    IPv6 support                              : enabled

